### PR TITLE
fix odometry issue in diff_drive_controller

### LINKED ISSUE #286: Wrong integration of velocity feedback in diff_drive_controller
See https://github.com/ros-controls/ros2_controllers/pull/260#discussion_r787651764
Thanks to @wmmc88 for spotting this bug.

### PR DESCRIPTION
**Pull request by roncapat**
Fixes #286 and the issue introduced in #260.

### CODE REVIEW

**Comment by roncapat** [comment]
Some considerations:
updatefromVelocity(9 takes as input not velocities, but position deltas... 
Lots of solutions here to transform a bit the Odometry API, but my idea is:
 - rename updateFromVelocity() to updateFromPositionDeltas() 
 - rename update() to updateFromPosition()
 - then, add a proxy method updateFromVelocity() that keeps the same parameters and internally computes `dt` to avoid to explicitly multiply by `update_dt.seconds()` as this patch currently does

Or:
 - change the set of parameters for the different odometry.hpp methods, as proposed in #271.
 - I believe that doing this will make the Odometry class less robust though. For now, it is something that integrates feedback readings at specified (through parameter `time`) timepoints. Passing instead time deltas would remove some nice semantic... No way to check if for example the passed timepoint follows or precedes the previously provided timepoint. and of course, modifying the API to pass both (time and time delta) would be redundant.

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -178,6 +178,9 @@ controller_interface::return_type DiffDriveController::update(
   double & linear_command = command.twist.linear.x;
   double & angular_command = command.twist.angular.z;
 
+  const auto update_dt = current_time - previous_update_timestamp_;
+  previous_update_timestamp_ = current_time;
```
[the remark this answers is not in the document]
Comment (roncapat) [reply diff_drive_controller/src/diff_drive_controller.cpp:182]: Sure! I just moved around the two pre-existing lines of code, but this proposed change is a nice improvement nevertheless.

**Comment by roncapat** [comment]
I would go with the renaming scheme I proposed if it's ok for you.

File: diff_drive_controller/src/diff_drive_controller.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmagyar) [reply diff_drive_controller/src/diff_drive_controller.cpp:182]: I'd argue to simply use the `period` passed to `update()`. It's the same information, already computed for you. These two lines could be reduced to
```suggestion
  const auto update_dt = period;
```
(plus you need to uncomment the argument above)

[the remark this answers is not in the document]
Comment (roncapat) [reply diff_drive_controller/src/diff_drive_controller.cpp:182]: Even better, but maybe we can just discard the whole `update_dt` definition and use `period` instead in code of the function. Would it be okay for you?

[the remark this answers is not in the document]
Comment (roncapat) [reply diff_drive_controller/src/diff_drive_controller.cpp]: Pushed [304c09c](https://github.com/ros-controls/ros2_controllers/pull/331/commits/304c09cd9f19015e907fe2243d43994cdbf4bf77) to simplify time variables usage in the code

**Review by bmagyar — APPROVED** [review]
This is already solving a problem which we need fixed. I'll go ahead with merging and @roncapat please submit a second PR with the renaming suggestions.

**Pull request merged**